File: HCS.Core/Domain/Provider.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Text;

namespace HCS.Core.Domain
{
    public class Provider
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int LocationId { get; set; }
        public Location Location { get; set; }
        public ICollection<ProvidedUtility> ProvidedUtilities { get; set; }

        public Provider()
        {
            ProvidedUtilities = new Collection<ProvidedUtility>();
        }
    }
}

File: HCS.Client/ClientApp/app/models/provider.ts
import { KeyValuePair } from "./key_value_pair";
import { ProvidedUtility } from "./provided-utility";

export class Provider {
    id: number;
    name: string;
    location: KeyValuePair | null;
    providedUtilities: ProvidedUtility[];
    constructor() {
        this.providedUtilities = [];
    }
}

export interface SaveProvider {
    id: number;
    name: string;
    locationId: number;
    providedUtilities: number[];
}